graphql_compiler/compiler/compiler_frontend.py

```diff
@@ -812,10 +813,92 @@ def _preprocess_graphql_string(graphql_string):
     return graphql_string + '\n'
 
 
+def _validate_schema_and_ast(schema, ast):
+    """Validate the supplied graphql schema and ast.
+
+    This method wraps around graphql-core's validation to enforce a stricter requirement of the
+    schema -- all directives supported by the compiler must be declared by the schema, regardless of
+    whether each directive is used in the query or not.
+
+    Args:
+        schema: GraphQL schema object, created using the GraphQL library
+        ast: abstract syntax tree representation of a graphql query
+
+    Returns:
+        list containing schema and/or query validation errors
+    """
+    core_graphql_errors = validate(schema, ast)
+
+    # The following directives appear in the core-graphql library, but are not supported by the
+    # graphql compiler.
+    unsupported_default_directives = frozenset([
+        frozenset([
+            'include',
+            frozenset(['FIELD', 'FRAGMENT_SPREAD', 'INLINE_FRAGMENT']),
+            frozenset(['if'])
+        ]),
+        frozenset([
+            'skip',
+            frozenset(['FIELD', 'FRAGMENT_SPREAD', 'INLINE_FRAGMENT']),
+            frozenset(['if'])
+        ]),
+        frozenset([
+            'deprecated',
+            frozenset(['ENUM_VALUE', 'FIELD_DEFINITION']),
+            frozenset(['reason'])
+        ])
+    ])
+
+    # Extract name, locations and args keys in order to compare schema directives and directives
+    # which are supported by the graphql compiler.
+
+    # Directives expected by the graphql compiler.
+    expected_directives = {
+        frozenset([
+            directive.name,
+            frozenset(directive.locations),
+            frozenset(six.viewkeys(directive.args))
+        ])
+        for directive in DIRECTIVES
+    }
+
+    # Graphql-core automatically injects default directives into the schema, regardless of whether
+    # the schema supports said directives. Hence, while the directives contained in
+    # unsupported_default_directives are incompatible with the graphql-compiler, we still expect
+    # them in the parsed schema string.
+    expected_directives.update(unsupported_default_directives)
```

Comment:
So the compiler doesn't actually care about whether the `unsupported_default_directives` appear in the schema or not -- if a future version of the core GraphQL library happens to remove them, the compiler won't mind. What do you think about making the code tolerate the appearance of these directives, but not require it?

So, for example, if the schema contains all the compiler-required directives, plus `deprecated` but none of the other default directives, the compiler could accept that.